eZ/Bundle/EzPublishLegacySearchEngineBundle/Command/CreateIndexCommand.php

```diff
@@ -0,0 +1,175 @@
+<?php
+
+/**
+ * This file is part of the eZ Publish Kernel package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Bundle\EzPublishLegacySearchEngineBundle\Command;
+
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use eZ\Publish\SPI\Persistence\Content;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Helper\ProgressHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Input\InputArgument;
+use eZ\Publish\SPI\Persistence\Content\ContentInfo;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\Core\Search\Legacy\Content\Handler as SearchHandler;
+use RuntimeException;
+use PDO;
+
+/**
+ * Console command ezplatform:create_sql_search_index indexes content objects for legacy search engine.
+ */
+class CreateIndexCommand extends ContainerAwareCommand
+{
+    /**
+     * @var \eZ\Publish\Core\Search\Legacy\Content\Handler
+     */
+    private $searchHandler;
+
+    /**
+     * @var \eZ\Publish\SPI\Persistence\Handler
+     */
+    private $persistenceHandler;
+
+    /**
+     * @var \eZ\Publish\Core\Persistence\Database\DatabaseHandler
+     */
+    private $databaseHandler;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * Initialize objects required by {@see execute()}.
+     *
+     * @param InputInterface $input
+     * @param OutputInterface $output
+     */
+    public function initialize(InputInterface $input, OutputInterface $output)
+    {
+        parent::initialize($input, $output);
+        $this->logger = $this->getContainer()->get('logger');
+        $this->searchHandler = $this->getContainer()->get('ezpublish.spi.search');
+        $this->persistenceHandler = $this->getContainer()->get('ezpublish.api.persistence_handler');
+        $this->databaseHandler = $this->getContainer()->get('ezpublish.connection');
+
+        if (!$this->searchHandler instanceof SearchHandler) {
+            throw new RuntimeException(
+                'Expected to find Search Handler but found something else.'
```

Comment:
You should mention Legacy Search Engine, the point is to have a helpful message if something is misconfigured.
See https://github.com/ezsystems/ezplatform-solr-search-engine/blob/master/bundle/Command/SolrCreateIndexCommand.php#L48-L51.